frame/babe/src/lib.rs

```diff
@@ -486,6 +490,8 @@ impl<T: Config> Module<T> {
 		Self::deposit_consensus(ConsensusLog::NextEpochData(next_epoch));
 
 		if let Some(next_config) = NextEpochConfig::take() {
+			let config: BabeEpochConfiguration = next_config.clone().into();
+			EpochConfig::put(config);
```

Comment:
Assuming that this function is enacting epoch `N`, this is the config for `N+1`, which we announce in advance.